go/libraries/doltcore/sqle/dtables/commit_diff_table.go

```diff
@@ -244,50 +244,57 @@ func (dt *CommitDiffTable) rootValForFilter(ctx *sql.Context, eqFilter *expressi
 func (dt *CommitDiffTable) HandledFilters(filters []sql.Expression) []sql.Expression {
 	var commitFilters []sql.Expression
 	for _, filter := range filters {
-		isCommitFilter := false
-
-		if eqFilter, isEquality := filter.(*expression.Equals); isEquality {
-			for _, e := range []sql.Expression{eqFilter.Left(), eqFilter.Right()} {
-				if val, ok := e.(*expression.GetField); ok {
-					switch strings.ToLower(val.Name()) {
-					case toCommit:
-						if dt.toCommitFilter != nil {
-							dt.requiredFilterErr = ErrExactlyOneToCommit
-						}
-
-						isCommitFilter = true
-						dt.toCommitFilter = eqFilter
-					case fromCommit:
-						if dt.fromCommitFilter != nil {
-							dt.requiredFilterErr = ErrExactlyOneFromCommit
-						}
-
-						isCommitFilter = true
-						dt.fromCommitFilter = eqFilter
-					}
-				}
-			}
+		eqFilter, isEquality := filter.(*expression.Equals)
+		if eqFilter == nil || !isEquality {
+			continue
 		}
-
-		if isCommitFilter {
-			commitFilters = append(commitFilters, filter)
+		for _, e := range []sql.Expression{eqFilter.Left(), eqFilter.Right()} {
+			val, ok := e.(*expression.GetField)
+			if !ok {
+				continue
+			}
+			switch strings.ToLower(val.Name()) {
+			case toCommit, fromCommit:
+				commitFilters = append(commitFilters, filter)
+			}
 		}
 	}
-
 	return commitFilters
 }
 
 // Filters returns the list of filters that are applied to this table.
 func (dt *CommitDiffTable) Filters() []sql.Expression {
-	if dt.toCommitFilter == nil || dt.fromCommitFilter == nil {
-		return nil
```

Comment:
actually this probably isn't OK, can't return nil filters, have to add all non-nils to a return list